src/components/TodoList/TodoList.tsx

```diff
@@ -1,210 +1,110 @@
+  let filteredTodos = todos;
+
+  if (trimQuery) {
+    // eslint-disable-next-line max-len
+    filteredTodos = filteredTodos.filter((todo) => todo.title.toLowerCase().includes(trimQuery.toLowerCase()));
+  }
+
+  switch (status) {
+    case 'active':
+      return filteredTodos.filter((todo) => !todo.completed);
+    case 'completed':
+      return filteredTodos.filter((todo) => todo.completed);
+    default:
+      return filteredTodos;
+  }
+};
 
-          <tr data-cy="todo">
-            <td className="is-vcentered">10</td>
+export const TodoList: React.FC<TodoListProps> = ({ todos, currentTodo }) => {
+  const dispatch = useAppDispatch();
+  const { query, status } = useAppSelector((state) => state.filter);
 
-            <td className="is-vcentered">
-              <span className="icon" data-cy="iconCompleted">
-                <i className="fas fa-check" />
-              </span>
-            </td>
+  const filteredTodos = useMemo(
+    () => getFilteredTodos(query, status, todos),
+    [todos, query, status],
+  );
 
-            <td className="is-vcentered is-expanded">
-              <p className="has-text-success">illo est ratione doloremque quia maiores aut</p>
-            </td>
+  return (
+    <>
+      {filteredTodos.length > 0 ? (
+        <table className="table is-narrow is-fullwidth">
+          <thead>
+            <tr>
+              <th>#</th>
 
-            <td className="has-text-right is-vcentered">
-              <button data-cy="selectButton" className="button" type="button">
+              <th>
                 <span className="icon">
-                  <i className="far fa-eye" />
+                  <i className="fas fa-check" />
                 </span>
-              </button>
-            </td>
-          </tr>
-        </tbody>
-      </table>
+              </th>
+
+              <th>Title</th>
+              <th> </th>
+            </tr>
+          </thead>
+
+          <tbody>
+            {filteredTodos.map((todo) => (
+              <tr data-cy="todo" key={todo.id}>
+                <td className="is-vcentered">{todo.id}</td>
+                <td className="is-vcentered">
+                  {todo.completed && (
+                    <span className="icon" data-cy="iconCompleted">
+                      <i className="fas fa-check" />
+                    </span>
+                  )}
+                </td>
+
+                <td className="is-vcentered is-expanded">
+                  <p
+                    className={
+                      todo.completed ? 'has-text-success' : 'has-text-danger'
+                    }
+                  >
+                    {todo.title}
+                  </p>
+                </td>
+
+                <td className="has-text-right is-vcentered">
+                  <button
+                    data-cy="selectButton"
+                    className="button"
+                    type="button"
+                    onClick={() => dispatch(actions.setTodo(todo))}
```

Comment:
you can move this logic to the top to separate function and use here as a callback:

```
const setTodo = () => dispatch(
    actions.setTodo(todo)
);
```

`onClick={setTodo}`